File: services/common/tracing.py
"""链路追踪: X-Trace-Id 注入与透传"""
import uuid
from typing import Optional

from fastapi import Request


def get_or_generate_trace_id(request: Request, prefix: str = "daduhe") -> str:
    """从请求Header提取trace_id，不存在则生成"""
    trace_id = request.headers.get("X-Trace-Id")
    if not trace_id:
        trace_id = f"{prefix}-{uuid.uuid4()}"
    return trace_id


def generate_trace_id(prefix: str = "daduhe") -> str:
    return f"{prefix}-{uuid.uuid4()}"

File: services/rule-extractor/src/common/tracing.ts
import { Request, Response, NextFunction } from "express";
import { v4 as uuidv4 } from "uuid";

const TRACE_HEADER = "X-Trace-Id";
const PREFIX = "rule-extractor";

/**
 * X-Trace-Id 注入与透传中间件。
 * 外部请求进入时自动生成，调用下游时透传收到的 trace_id。
 */
export function tracingMiddleware(
  req: Request,
  res: Response,
  next: NextFunction
): void {
  const traceId = (req.headers[TRACE_HEADER.toLowerCase()] as string) || `${PREFIX}-${uuidv4()}`;
  req.headers[TRACE_HEADER.toLowerCase()] = traceId;
  res.setHeader(TRACE_HEADER, traceId);
  next();
}
